### PR TITLE
[zh-cn]: sync translation for File System API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Related issues and pull requests

depends on https://github.com/mdn/content/pull/35772

### CODE REVIEW

File: files/zh-cn/web/api/filesystemhandle/querypermission/index.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: FileSystemHandle：queryPermission() 方法
 slug: Web/API/FileSystemHandle/queryPermission
 l10n:
-  sourceCommit: be3c45cd7a4d5c04139eceae10f7368251cdca64
+  sourceCommit: a83790f596feb4abac01cbffbf45fc8c1e117d97
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/filesystemhandle/querypermission/index.md:5]:
```suggestion
  sourceCommit: 4e8bc4593e38b3902430fa701a6256c95d7bbbdc
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 2]
```diff
@@ -15,11 +15,23 @@ l10n:
...
 
-无。
+- `options` {{optional_inline}}
+
+  - : 一个具有以下属性的对象：
+
+    - `mode` {{optional_inline}} {{non-standard_inline}}
+      - : 指定访问句柄的锁定模式的字符串。默认值为 `"readwrite"`。可能的值包括：
+        - `"read-only"`
+          - : 可以同时在一个文件上打开多个 `FileSystemSyncAccessHandle` 对象（例如，在多个选项卡中使用同一个应用时），前提是它们都以 `"read-only"` 模式打开。打开后，可以在句柄上调用类似读取的方法——{{domxref("FileSystemSyncAccessHandle.read", "read()")}}、{{domxref("FileSystemSyncAccessHandle.getSize", "getSize()")}} 和 {{domxref("FileSystemSyncAccessHandle.close", "close()")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:30]:
```suggestion
          - : 可以同时在一个文件上打开多个 `FileSystemSyncAccessHandle` 对象（例如，在多个标签页中使用同一个应用时），前提是它们都以 `"read-only"` 模式打开。打开后，可以在句柄上调用类似读取的方法——{{domxref("FileSystemSyncAccessHandle.read", "read()")}}、{{domxref("FileSystemSyncAccessHandle.getSize", "getSize()")}} 和 {{domxref("FileSystemSyncAccessHandle.close", "close()")}}。
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 3]
```diff
@@ -58,6 +72,137 @@ onmessage = async (e) => {
 };
 ```
 
+### 带有 `mode` 选项的完整示例
+
+我们的 [`createSyncAccessHandle()` 模式测试](https://createsyncaccesshandle-mode-test.glitch.me/) 示例提供了一个 {{htmlelement("input")}} 字段来输入文本，以及两个按钮——一个用于将输入的文本写入原始私有文件系统中的文件末尾，另一个用于在文件太满时清空文件。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:77]:
```suggestion
我们的 [`createSyncAccessHandle()` 模式测试](https://createsyncaccesshandle-mode-test.glitch.me/)示例提供了一个 {{htmlelement("input")}} 字段来输入文本，以及两个按钮——一个用于将输入的文本写入原始私有文件系统中的文件末尾，另一个用于在文件太满时清空文件。
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 4]
```diff
@@ -58,6 +72,137 @@ onmessage = async (e) => {
...
+
+两个 {{htmlelement("button")}} 元素和文本 {{htmlelement("input")}} 字段如下所示：
+
+```html
+<ol>
+  <li>
+    <label for="filetext">输入要写入文件的文本：</label>
+    <input type="text" id="filetext" name="filetext" />
+  </li>
+  <li>将你的文本写入文件：<button class="write">撰写文本</button></li>
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:93]:
```suggestion
  <li>将你的文本写入文件：<button class="write">写入文本</button></li>
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 5]
```diff
@@ -58,6 +72,137 @@ onmessage = async (e) => {
...
+  <li>
+    如果文件太满，则清空该文件：
+    <button class="empty">空文件</button>
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:96]:
```suggestion
    如果文件太满，则清空该文件：<button class="empty">清空文件</button>
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 6]
```diff
@@ -58,6 +72,137 @@ onmessage = async (e) => {
...
+    size = accessHandle.getSize();
+
+    // 准备文件长度的数据视图
+    const dataView = new DataView(new ArrayBuffer(size));
+
+    // 将整个文件读入数据视图
+    accessHandle.read(dataView, { at: 0 });
+
+    // 将当前文件内容记录到控制台
+    console.log("File contents: " + textDecoder.decode(dataView));
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:195]:
```suggestion
    console.log("文件内容：" + textDecoder.decode(dataView));
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 7]
```diff
@@ -26,6 +26,12 @@ createWritable(options)
 
     - `keepExistingData` {{optional_inline}}
       - : {{jsxref('Boolean', '布尔值', '', 'nocode')}}，默认为 `false`。当设为 `true` 时，如果文件存在，则现将现有文件的内容复制到临时文件，否则临时文件初始时内容为空。
+    - `mode` {{optional_inline}} {{non-standard_inline}}
+      - : 指定可写文件流的锁定模式的字符串。默认值为 `"siloed"`。可能的值包括：
+        - `"exclusive"`
+          - : 只能打开一个 `FileSystemWritableFileStream` 写入器。在第一个写入器关闭之前尝试打开后续写入器会导致抛出 `NoModificationAllowedError` 异常。
+        - `"siloed"`
+          - : 可以同时打开多个 `FileSystemWritableFileStream` 写入器，每个写入器都有自己的交换文件，例如在多个选项卡中使用同一个应用时。最后打开的写入器会写入其数据，因为每个写入器关闭时都会刷新数据。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:34]:
```suggestion
          - : 可以同时打开多个 `FileSystemWritableFileStream` 写入器，每个写入器都有自己的交换文件，例如在多个标签页中使用同一个应用时。最后打开的写入器会写入其数据，因为每个写入器关闭时都会刷新数据。
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 8]
```diff
@@ -59,6 +67,118 @@ async function writeFile(fileHandle, contents) {
 }
 ```
 
+### 通过选项扩展用途
+
+我们的 [`createWritable()` 模式测试](https://createwritable-mode-test.glitch.me/)示例提供了一个 {{htmlelement("button")}} 来选择要写入的文件，一个文本 {{htmlelement("input")}} 字段，你可以在其中输入一些要写入文件的文本，以及第二个 `<button>` 来将文本写入文件。
+
+在上面的演示中，尝试选择文件系统上的文本文件（或输入新文件名），在输入字段中输入一些文本，然后将文本写入文件。打开文件系统上的文件以检查写入是否成功。
+
+此外，尝试同时在两个浏览器选项卡中打开页面。在第一个标签页中选择要写入的文件，然后立即尝试在第二个标签页中选择要写入的同一文件。你应该会收到一条错误消息，因为我们在 `createWritable()` 调用中设置了 `mode: "exclusive"`。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:76]:
```suggestion
此外，尝试同时在两个浏览器标签页中打开页面。在第一个标签页中选择要写入的文件，然后立即尝试在第二个标签页中选择要写入的同一文件。你应该会收到一条错误消息，因为我们在 `createWritable()` 调用中设置了 `mode: "exclusive"`。
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 9]
```diff
@@ -59,6 +67,118 @@ async function writeFile(fileHandle, contents) {
...
+
+下面我们将探索代码。
+
+#### HTML
+
+The two {{htmlelement("button")}} elements and text {{htmlelement("input")}} field look like this:
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:82]:
```suggestion
两个 {{htmlelement("button")}} 元素和文本 {{htmlelement("input")}} 字段如下所示：
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 10]
```diff
@@ -59,6 +67,118 @@ async function writeFile(fileHandle, contents) {
...
+```html
+<ol>
+  <li>选择要写入的文件：<button class="select">选择文件</button></li>
+  <li>
+    <label for="filetext">输入要写入文件的文本：</label>
+    <input type="text" id="filetext" name="filetext" disabled />
+  </li>
+  <li>
+    将你的文本写入文件：
+    <button class="write" disabled>撰写文本</button>
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:93]:
```suggestion
    将你的文本写入文件：<button class="write" disabled>写入文本</button>
```

File: files/zh-cn/web/api/filesystemhandle/issameentry/index.md [hunk 11]
```diff
@@ -26,12 +26,16 @@ isSameEntry(fileSystemHandle)
 
 ## 示例
 
-以下函数将单个条目与条目数组进行比对，返回包含不相符条目的新数组。
+以下函数将单个条目与条目数组进行比对，返回兑现包含不相符条目的新数组的 {{jsxref("Promise")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/issameentry/index.md:29]:
```suggestion
以下函数将单个条目与条目数组进行比对，并返回一个 {{jsxref("Promise")}}，其会兑现包含不相符条目的新数组。
```

File: files/zh-cn/web/api/filesystemhandle/querypermission/index.md [hunk 12]
```diff
@@ -27,9 +27,9 @@ queryPermission(descriptor)
 
 ### 返回值
 
-一个兑现 {{domxref("PermissionStatus")}} 对象的 {{jsxref("Promise")}}。
+一个兑现为 {{domxref('PermissionStatus.state')}} 的 {{jsxref("Promise")}}，其值为 `'granted'`、`'denied'` 或 `'prompt'`。它也可能因出现以下例外情况之一而被拒绝。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/querypermission/index.md:30]:
```suggestion
一个兑现为 {{domxref('PermissionStatus.state')}} 的 {{jsxref("Promise")}}，其值为 `'granted'`、`'denied'` 或 `'prompt'`。它也可能因出现以下异常而被拒绝。
```

File: files/zh-cn/web/api/filesystemhandle/querypermission/index.md [hunk 13]
```diff
@@ -27,9 +27,9 @@ queryPermission(descriptor)
...
 
-如果 {{domxref('PermissionStatus.state')}} 返回了“prompt”，则站点必须在对句柄进行任何操作前调用 `requestPermission()` 请求授权。如果此方法返回了“denied”，则任何操作都会被拒绝。从本地文件系统句柄构造器返回的句柄通常会在初始时对只读权限状态返回“granted”。但是，除开用户收回了权限的情况，从 IndexedDB 获取的句柄也有可能会返回“prompt”。
+如果其兑现为“prompt”，则网站必须先调用 `requestPermission()`，然后才能对句柄执行任何操作。如果其兑现为“denied”，则任何操作都将被拒绝。通常，本地文件系统句柄工厂返回的句柄最初将解析为“granted”作为其读取权限状态。但是，除了通过用户撤销权限之外，从 IndexedDB 检索到的句柄也可能解析为“prompt”。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/querypermission/index.md:32]:
```suggestion
如果其兑现为“prompt”，则网站必须先调用 `requestPermission()`，然后才能对句柄执行任何操作。如果其兑现为“denied”，则任何操作都将被拒绝。通常，本地文件系统句柄工厂返回的句柄最初将兑现“granted”作为其读取权限状态。但是，除了通过用户撤销权限之外，从 IndexedDB 检索到的句柄也可能兑现为“prompt”。
```

File: files/zh-cn/web/api/filesystemhandle/requestpermission/index.md [hunk 14]
```diff
@@ -27,12 +27,20 @@ requestPermission(descriptor)
 
 ### 返回值
 
-一个兑现 {{domxref("PermissionStatus")}} 对象的 {{jsxref("Promise")}}。
+一个兑现为 {{domxref('PermissionStatus.state')}} 的 {{jsxref("Promise")}}，其值为 `'granted'`、`'denied'` 或 `'prompt'`。它也可能因出现以下例外情况之一而被拒绝。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/requestpermission/index.md:30]:
```suggestion
一个兑现为 {{domxref('PermissionStatus.state')}} 的 {{jsxref("Promise")}}，其值为 `'granted'`、`'denied'` 或 `'prompt'`。它也可能因出现以下异常而被拒绝。
```

File: files/zh-cn/web/api/filesystemhandle/requestpermission/index.md [hunk 15]
```diff
@@ -27,12 +27,20 @@ requestPermission(descriptor)
...
 
 ### 异常
 
 - {{jsxref("TypeError")}}
   - : 如果没有指定参数或者 `mode` 的值不是 `'read'` 或 `'readwrite'`，则抛出此异常。
+- `SecurityError` {{domxref("DOMException")}}
+  - : 在下列情况之一发生时抛出：
+    - 该方法是在与顶级上下文不[同源](/zh-CN/docs/Web/Security/Same-origin_policy)的上下文中调用的（例如跨源 iframe）.
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/requestpermission/index.md:38]:
```suggestion
    - 该方法是在与顶级上下文不[同源](/zh-CN/docs/Web/Security/Same-origin_policy)的上下文（例如跨源 iframe）中调用的。
```

File: files/zh-cn/web/api/filesystemhandle/requestpermission/index.md [hunk 16]
```diff
@@ -27,12 +27,20 @@ requestPermission(descriptor)
...
+    - 没有瞬态用户激活，例如按下按钮。这包括当句柄处于无法使用用户激活的非 Window 上下文中时，例如 Worker.
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/requestpermission/index.md:39]:
```suggestion
    - 没有瞬态用户激活，例如按下按钮。这包括当句柄处于无法使用用户激活的非 Window 上下文（例如 Worker）中时。
```

File: files/zh-cn/web/api/filesystemhandle/requestpermission/index.md [hunk 17]
```diff
@@ -27,12 +27,20 @@ requestPermission(descriptor)
...
+
+## 安全性
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemhandle/requestpermission/index.md:41]: 用“安全”的更多，我们要统一一下

```suggestion
## 安全
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 18]
```diff
@@ -58,6 +72,137 @@ onmessage = async (e) => {
 };
 ```
 
+### 带有 `mode` 选项的完整示例
+
+我们的 [`createSyncAccessHandle()` 模式测试](https://createsyncaccesshandle-mode-test.glitch.me/) 示例提供了一个 {{htmlelement("input")}} 字段来输入文本，以及两个按钮——一个用于将输入的文本写入原始私有文件系统中的文件末尾，另一个用于在文件太满时清空文件。
+
+尝试探索上面的演示，打开浏览器开发者控制台，以便你可以看到正在发生的事情。如果你尝试在多个浏览器选项卡中打开演示，你会发现可以同时打开多个句柄以同时写入文件。这是因为在 `createSyncAccessHandle()` 调用上设置了 `mode: "readwrite-unsafe"`。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:79]:
```suggestion
尝试探索上面的演示，打开浏览器开发者控制台，以便你可以看到正在发生的事情。如果你尝试在多个浏览器标签页中打开演示，你会发现可以同时打开多个句柄以同时写入文件。这是因为在 `createSyncAccessHandle()` 调用上设置了 `mode: "readwrite-unsafe"`。
```

**Review by yin1999 — APPROVED** [review]
Thanks @skyclouds2001

**Pull request merged**